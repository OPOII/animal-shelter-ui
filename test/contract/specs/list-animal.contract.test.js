import {Matchers} from '@pact-foundation/pact';
import {AnimalController} from '../../../controllers';
import {provider} from '../config/initPact';

describe('Animal Service', () => {
    describe('When a request to list all animals is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'there are animals',
                uponReceiving: 'a request to list all animals',
                withRequest: {
                    method: 'GET',
                    path: '/animals'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike(
                        {
                            name: Matchers.like('manchas'),
                            breed: Matchers.like("Bengali"),
                            gender: Matchers.term(
                                {generate: "Female", matcher: "Female|Male"}
                            ),
                            isVaccinated: Matchers.boolean(true),
                            vaccines: Matchers.eachLike(
                                [
                                    "rabia"
                                ],
                                {min: 1}
                            )
                        },
                        {min: 2}
                    )
                }
            });
        });

        it('should return the correct data', async () => {
            const response = await AnimalController.list();
            expect(response.data).toMatchSnapshot();
            
            await provider.verify()
        });

        afterAll(async() => { await provider.finalize();
        });
    });
});