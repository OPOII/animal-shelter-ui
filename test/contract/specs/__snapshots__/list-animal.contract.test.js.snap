// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to list all animals is made should return the correct data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "isVaccinated": true,
    "name": "manchas",
    "vaccines": Array [
      Array [
        "rabia",
      ],
    ],
  },
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "isVaccinated": true,
    "name": "manchas",
    "vaccines": Array [
      Array [
        "rabia",
      ],
    ],
  },
]
`;
